src/test/java/io/vlingo/hello/resource/GreetingResourceTest.java

```diff
@@ -0,0 +1,169 @@
+package io.vlingo.hello.resource;
+
+import io.vlingo.hello.infra.DescriptionData;
+import io.vlingo.hello.infra.GreetingData;
+import io.vlingo.hello.infra.MessageData;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.StringRegularExpression.matchesRegex;
+
+public class GreetingResourceTest extends ResourceTestCase {
+
+  @Test
+  public void itCreatesNewGreeting() {
+    givenJsonClient()
+        .body(greetingData("Message", "Description"))
+        .when()
+        .post("/greetings")
+        .then()
+        .statusCode(201)
+        .header("Location", matchesRegex("/greetings/[0-9]+"))
+        .body(
+            "id", notNullValue(),
+            "message", equalTo("Message"),
+            "description", equalTo("Description"),
+            "messageChangedCount", equalTo(0),
+            "descriptionChangedCount", equalTo(0)
+        );
+  }
+
+  @Test
+  public void itGetsAnExistingGreeting() {
+    String location = givenGreetingWasCreated(greetingData("Message", "Description"));
+
+    givenJsonClient()
+        .when()
+        .get(location)
+        .then()
+        .statusCode(200)
+        .body(
+            "id", equalTo(locationToId(location)),
+            "message", equalTo("Message"),
+            "description", equalTo("Description"),
+            "messageChangedCount", equalTo(0),
+            "descriptionChangedCount", equalTo(0)
+        );
+  }
+
+  @Test
+  public void itReturns404IfGreetingIsNotFound() {
+    givenJsonClient()
+        .when()
+        .get("/greetings/42")
+        .then()
+        .statusCode(404);
+  }
+
+  @Test
+  public void itGetsExistingGreetings() {
+    givenGreetingWasCreated(greetingData("Message", "Description"));
+
+    givenJsonClient()
+        .when()
+        .get("/greetings")
+        .then()
+        .statusCode(200)
+        .body(
+            "[0].id", notNullValue(),
+            "[0].message", equalTo("Message"),
+            "[0].description", equalTo("Description"),
+            "[0].messageChangedCount", equalTo(0),
+            "[0].descriptionChangedCount", equalTo(0)
+        );
+  }
+
+  @Test
+  public void itChangesTheMessage() {
+    String id = locationToId(
+        givenGreetingWasCreated(greetingData("Message", "Description"))
+    );
+
+    givenJsonClient()
+        .body(messageData("New Message"))
+        .when()
+        .patch(String.format("/greetings/%s/message", id))
+        .then()
+        .statusCode(200)
+        .body(
+            "id", equalTo(id),
+            "message", equalTo("New Message"),
+            "description", equalTo("Description"),
+            "messageChangedCount", equalTo(1),
+            "descriptionChangedCount", equalTo(0)
+        );
+  }
+
+  @Test
+  @Ignore
```

Comment:
Two tests are ignored as https://github.com/vlingo/vlingo-helloworld/issues/2 wasn't fully fixed in this repository yet.

Reply:
This is really nice. Thanks!

Slack discussion summarized:

(1) The demo should be using `Stage.maybeActorOf()`. That will return a `Completes<Optional<SomeProtocol>>`. And I think would be be best to change all demo code and all production code (currently our own `vlingo-schemata` if not already done) to use `Stage.maybeActorOf()`. Inside the `andThen()` or `andThenTo()` we can map on the `Optional`. 

(2) We could also make `DirectoryScanner` support our `Outcome<F,S>` which is like `Try` and `Either`. We could make the `Completes<Outcome<F,S>>` surface on `Stage.tryActorOf()` and on `DirectoryScanner.tryActorOf()`. That would make it easier to branch inside the `andThen()` and `andThenTo()` functions without using `if-else` for failures and successes.

I think option 2 is best. Would you be interested in implementing that?

Reply:
I will take a look.